src/v2/view-builder/views/TerminalView.js

```diff
@@ -30,12 +31,31 @@ const GET_BACK_TO_SIGN_LINK_FLOWS = [
   REGISTRATION_NOT_ENABLED,
 ];
 
+const DEVICE_CODE_ERROR_KEYS = [
+  DEVICE_NOT_ACTIVATED_CONSENT_DENIED,
+  DEVICE_NOT_ACTIVATED_INTERNAL_ERROR
+];
+
+const DEVICE_CODE_FLOW_TERMINAL_KEYS = [
+  DEVICE_ACTIVATED,
+  ...DEVICE_CODE_ERROR_KEYS
+];
+
 const Body = BaseForm.extend({
   noButtonBar: true,
 
   postRender() {
     BaseForm.prototype.postRender.apply(this, arguments);
     this.$el.addClass('terminal-state');
+
+    // show device code terminal icons
+    if (this.options.appState.containsMessageWithI18nKey(DEVICE_CODE_FLOW_TERMINAL_KEYS)) {
+      const iconClass = this.options.appState.containsMessageWithI18nKey(DEVICE_ACTIVATED)
+        ? 'success-24-green' : 'error-24-red';
+      this.$('.o-form-head').before('<div class="device-code-terminal--icon-container">' +
```

Comment:
I think instead of manipulating the DOM here.. we can make it part of BaseForm template or
create a new terminal view for this case.. Aim is to restrict this TerminalView to get too much bloated over time.

Reply:
Created a follow up: OKTA-387077